src/components/HomePage/SearchArea/SearchArea.js

```diff
@@ -60,12 +82,33 @@ export const AdvancedSearchController = ({
         technologies,
     });
 
-    const submitForm = useCallback((e) => {
+    const submitForm = useCallback((e, updateUrl = true) => {
         if (e) e.preventDefault();
         searchOffers(SearchResultsConstants.INITIAL_LIMIT);
 
+        if (updateUrl) setUrlSearchValue(searchValue);
+
         if (onSubmit) onSubmit();
-    }, [onSubmit, searchOffers]);
+    }, [onSubmit, searchOffers, searchValue, setUrlSearchValue]);
+
+    useEffect(() => {
+        if (queryParams.jobMinDuration && queryParams.jobMaxDuration) {
+            setShowJobDurationSlider(true);
+            setJobDuration(null, [
+                parseInt(queryParams.jobMinDuration, 10),
+                parseInt(queryParams.jobMaxDuration, 10),
+            ]);
+        }
+
+        if (queryParams.jobType) setJobType(queryParams.jobType);
+        if (queryParams.fields) setFields(queryParams.fields);
+        if (queryParams.technologies) setTechs(queryParams.technologies);
+
+        if (queryParams.searchValue) {
+            setSearchValue(queryParams.searchValue);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
Is this fine?

Reply:
Yes, because we only want to process the URL values once. Maybe I misunderstood your question...